Properly check header before extracting the bearer token

gsmet opened the pull request:
Fixes #42591

gsmet committed:
Properly check header before extracting the bearer token

Fixes #42591

sberyozkin reviewed (approved):
Thanks @gsmet

sberyozkin commented on extensions/elytron-security-oauth2/runtime/src/main/java/io/quarkus/elytron/security/oauth2/runtime/auth/OAuth2AuthMechanism.java:
```diff
@@ -42,15 +44,17 @@ public class OAuth2AuthMechanism implements HttpAuthenticationMechanism {
-            return identityProviderManager
-                    .authenticate(new TokenAuthenticationRequest(new TokenCredential(bearerToken, "bearer")));
 
+        if (authHeader == null || !authHeader.startsWith(BEARER_PREFIX)) {
```
May be tests have some spaces around... It all looks fine

gsmet replied:
The test I checked is using `Bearer: <token>` with a colon which is AFAICS incorrect. I'll go fix it I suppose.

gsmet replied:
I added an additional commit, I'm in the middle of something and can't test my tree.

gsmet committed:
Fix ElytronOauth2ExtensionResourceTestCase

The header should be:
Authorization: Bearer <token>
and not
Authorization: Bearer: <token>

gsmet commented:
@sberyozkin I let you check that I’m correct regarding the tests. All the examples I could find didn’t have a colon.

sberyozkin commented:
Thanks @gsmet This extension was added by Loic long time ago, he did not like the idea of depending on smallrye-jwt. This extension is light weight indeed but it only supports the remote token introspection via Elytron AFAIK

sberyozkin commented:
Thanks @michalvavrik

sberyozkin commented:
@gsmet @michalvavrik, FYI, https://quarkus.io/guides/security-authentication-mechanisms#oidc-jwt-oauth2-comparison
